### PR TITLE
feat: Add PDF Splitter

### PR DESCRIPTION
**Pull request by holtskinner**
- Integrated PDF Splitter as implemented in https://github.com/GoogleCloudPlatform/document-ai-samples/tree/main/pdf-splitter-python

### CODE REVIEW

File: google/cloud/documentai_toolbox/wrappers/document.py [hunk 1]
```diff
@@ -344,3 +346,44 @@ def get_entity_by_type(self, target_type: str) -> List[Entity]:
...
+        output_files: List[str] = []
+        input_filename, input_extension = os.path.splitext(os.path.basename(pdf_path))
+        with Pdf.open(pdf_path) as f:
+            for entity in self.entities:
+                subdoc_type = entity.type_ or "subdoc"
+
+                if entity.start_page == entity.end_page:
+                    page_range = f"pg{entity.start_page + 1}"
+                else:
+                    page_range = f"pg{entity.start_page + 1}-{entity.end_page + 1}"
```
Comment (galz10) [inline google/cloud/documentai_toolbox/wrappers/document.py:371]: You don't have a resource that is actually testing this line, please make sure this line is tested

File: setup.py [hunk 2]
```diff
@@ -52,6 +52,7 @@
         "google-cloud-documentai >= 1.2.1, < 3.0.0dev",
         "google-cloud-storage >= 1.31.0, < 3.0.0dev",
         "numpy >= 1.18.1",
+        "pikepdf >= 6.2.9, < 8.0.0",
```
Comment (galz10) [inline setup.py:55]: Why did you use pikepdf over PyPDF2 ?

Comment (holtskinner) [reply setup.py:55]: This is the library that Matthew Carroll used for his original sample.
From my understanding, he went through a vetting process to decide what to use. https://github.com/GoogleCloudPlatform/document-ai-samples/tree/main/pdf-splitter-python

File: google/cloud/documentai_toolbox/wrappers/document.py (hunk 1 above)
Comment (holtskinner) [reply google/cloud/documentai_toolbox/wrappers/document.py:371]: Specifically testing that the correct pages are being split?
That makes sense

File: google/cloud/documentai_toolbox/wrappers/document.py [hunk 3]
```diff
@@ -344,3 +346,44 @@ def get_entity_by_type(self, target_type: str) -> List[Entity]:
 
         """
         return [entity for entity in self.entities if entity.type_ == target_type]
+
+    def split_pdf(self, pdf_path: str, output_path: str) -> List[str]:
```
[the remark this answers is not in the document]
Comment (holtskinner) [reply google/cloud/documentai_toolbox/wrappers/document.py:350]: The splitter/classifier output is in the Document object, and it uses the entity fields to determine split points

File: setup.py (hunk 2 above)
Comment (galz10) [inline setup.py:55]: Please add this import to testing folder

**Review by galz10 — APPROVED** [review]

**Pull request merged**